load_roomates_into_linelist.R

```diff
@@ -0,0 +1,138 @@
+        filter(!ufid %in% case_data$contact_ll_ufid) %>%
+        summarise(ufid = paste(ufid, collapse = ",")) %>%
+        # if summarise returns an empty row drop it
+        filter(ufid != "")
+
+      if (nrow(case_roomate_linelist_entry) > 0) {
+
+        read_result_data <- read_redcap_db(
+          redcap_uri = Sys.getenv("URI"),
+          token = Sys.getenv("RESULT_TOKEN"),
+          project_title = Sys.getenv("RESULT_PROJECT_TITLE"),
+          records_collapsed = case_roomate_linelist_entry$ufid,
+          forms = "uf_identifiers_and_determination",
+          batch_size = 100
+        )
+
+        if (length(read_result_data) > 2) {
+
+          result_data <- read_result_data$data
+
+          # Accepted phone format for REDcap
+          us_phones <- "^([2-9][0-8][0-9])?([2-9][0-9]{2})([0-9]{4})$"
+
+          case_linelist_upload <- result_data %>%
+            select(contact_ll_ufid = record_id,
+                   contact_ll_firstname = first_name,
+                   contact_ll_lastname = last_name,
+                   contact_ll_phone = phone_number,
+                   contact_ll_uf_housing = uf_housing,
+                   contact_ll_data_source = data_source,
+                   contact_ll_infection_area = infection_area) %>%
+            group_by(contact_ll_infection_area) %>%
+            mutate(
+              redcap_event_name = 'case_investigation_arm_1',
+              redcap_repeat_instrument = 'covid19_case_contact_line_list',
+              redcap_repeat_instance = row_number(),
+              # format phone for redcap upload
+              contact_ll_phone = if_else(
+                nchar(contact_ll_phone) == 11,
+                str_remove(contact_ll_phone, "1"),
+                as.character(contact_ll_phone)
+              ),
+              contact_ll_phone = if_else(
+                str_detect(contact_ll_phone, us_phones),
+                contact_ll_phone,
+                NA_character_
+              )) %>%
+            # get the case record_id that generated the linelist entry
+            left_join(case_data %>%
+                        select(record_id, case_infection_area),
+                      by = c("contact_ll_infection_area" = "case_infection_area")) %>%
+            select(record_id, redcap_event_name, redcap_repeat_instrument,
+                   redcap_repeat_instance, everything())
+
+          write_data_to_case <- upload_to_redcap_db(
+            df_to_upload = case_linelist_upload,
+            redcap_uri = Sys.getenv("URI"),
+            token = Sys.getenv("CASE_TOKEN"),
+            project_title = Sys.getenv("CASE_PROJECT_TITLE"))
+
+          if (length(write_data_to_case) > 2) {
+
+            email_subject <- paste("Case line list entries created", "|",
+                                   Sys.getenv("INSTANCE"), "|", script_run_time)
+
+            message <- paste("write_data_to_case$records_affected_count",
+                             "line-list entries written to ",
```

Comment:
```suggestion
            message <- paste(write_data_to_case$records_affected_count,
                             "line-list entries written to",
```

You didn't want to quote that variable and there was an extra space on the second line

Reply:
I'll send this in a commit with a couple other changes